workshops/chat/README.md

```diff
@@ -0,0 +1,258 @@
+# Chat
+
+Short link to this workshop: [https://workshops.hackclub.com/chat](https://workshops.hackclub.com/chat)
+
+---
+
+In this workshop we'll be implementing a real-time chat using Firebase. This workshop is a revival of the nowdefunct tutorial made by Firebase itself.
+
+**Table of Contents**
+
+- Part I: Set-up
+- Part II: The HTML File
+- Part III: The JS File
+- Part IV: Publishing and Sharing (and Chatting!)
+- Part V: Hacking
+
+## Part I: Set-up
+
+- Open up your C9 workspace
+- Make a new folder and name it `chat`
+- Make a new file within `chat` and name it `index.html`
+- Make another new file within `chat` and name it `main.js`
+- Add a reference to Firebase, by adding this to the **body** of `index.html`:
+
+  ```html
+  <script src="https://www.gstatic.com/firebasejs/live/3.0/firebase.js"></script>
+  ```
+
+- Add a reference to jQuery,by adding this after the previous line:
+
+  ```html
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
+  ```
+
+- Lastly, include the `main.js`, by adding this after the previous line:
+
+  ```html
+  <script src="main.js"></script>
+  ```
+
+## Part II: The HTML File
+
+- Add a div to store message history in the **body**, above the script tags
+
+```html
+<div id="messagesDiv"></div>
+```
+
+- put two inputs into the **body**, above the script tags
+
+```html
+<input type="text" id="nameInput" placeholder="Name">
+<input type="text" id="messageInput" placeholder="Message">
+```
+
+Save and open up the external live preview.
+
+## Part III: The JS File
+
+### Preparing to Use Firebase
+
+First we'll create an app on firebase.
+
+![](img/fb_create_project.gif)
+
+Let's click on "Add Firebase to your web app." We'll copy part of the code snippet and paste it into our `main.js`:
+
+![](img/fb_get_config.gif)
+
+Now we're in the dashboard again.
+
+![](img/TODO dashboard)
+
+We have to add the domain of our external live preview to Firebase, so that we can debug while building the app. We'll click on "Auth" in the sidebar, and then the "Sign-In Method" tab:
+
+![](img/TODO go to side bar "auth")
+![](img/TODO change tab to sign-in method)
```

Comment:
Before having the user do all this, I might let them run into the error by asking them to see the live preview, open the console, then observe the error.